src/vs/workbench/browser/layout.ts

```diff
@@ -48,33 +52,43 @@ interface ComputedStyles {
 /**
  * The workbench layout is responsible to lay out all parts that make the Monaco Workbench.
  */
-export class WorkbenchLayout implements IVerticalSashLayoutProvider {
+export class WorkbenchLayout implements IVerticalSashLayoutProvider, IHorizontalSashLayoutProvider {
 
-	private static sashWidthSettingsKey = 'workbench.sidebar.width';
+	private static sashXWidthSettingsKey = 'workbench.sidebar.width';
+	private static sashYHeightSettingsKey = 'workbench.panelpart.height';
 
 	private parent: Builder;
 	private workbenchContainer: Builder;
 	private activitybar: Part;
 	private editor: Part;
 	private sidebar: Part;
+	private panelPart: Part;
 	private statusbar: Part;
 	private quickopen: QuickOpenController;
 	private options: LayoutOptions;
 	private toUnbind: { (): void; }[];
 	private computedStyles: ComputedStyles;
-	private editorHeight: number;
 	private workbenchSize: Dimension;
-	private sash: Sash;
+	private sashX: Sash;
+	private sashY: Sash;
 	private startSidebarWidth: number;
 	private sidebarWidth: number;
+	private sidebarHeight: number;
+	private startPanelPartHeight: number;
```

Comment:
@isidorn again easier to read the thing as "panel" unless you actually talk about the panelPart class